File: frontend/src/api/qr.ts
import api from './client';

export interface QrGenerateRequest {
  merchant_id: string;
  amount: number;
  reference?: string;
  purpose?: string;
}

export interface QrGenerateResponse {
  qr_string: string;
  amount: number;
  reference: string | null;
  purpose: string | null;
}

export async function generateQr(data: QrGenerateRequest): Promise<QrGenerateResponse> {
  const { data: result } = await api.post<QrGenerateResponse>('/api/qr/generate', data);
  return result;
}

File: backend/app/schemas/qr.py
"""Schemas for QR generation endpoint."""

from pydantic import BaseModel


class QrGenerateRequest(BaseModel):
    merchant_id: str
    amount: int
    reference: str | None = None
    purpose: str | None = None


class QrGenerateResponse(BaseModel):
    qr_string: str
    amount: int
    reference: str | None = None
    purpose: str | None = None
